backends/candle/src/models/flash_qwen3.rs

```diff
@@ -0,0 +1,507 @@
+            [
+                input_shape,
+                &[self.num_attention_heads, self.attention_head_size],
+            ]
+            .concat(),
+        )?;
+        let k = k.reshape(
+            [
+                input_shape,
+                &[self.num_key_value_heads, self.attention_head_size],
+            ]
+            .concat(),
+        )?;
+        let v = v.reshape(
+            [
+                input_shape,
+                &[self.num_key_value_heads, self.attention_head_size],
+            ]
+            .concat(),
+        )?;
+
+        // Apply normalization layers
+        let (q, _res) = self.q_norm.forward(&q, None)?;
+        let (k, _res) = self.k_norm.forward(&k, None)?;
+
+        apply_rotary_inplace(&q, &k, &cos, &sin, true)?;
+
+        let attention = flash_attn_varlen(
+            &q,
+            &k,
+            &v,
+            None,
+            cu_seqlens,
+            cu_seqlens,
+            max_s,
+            max_s,
+            self.softmax_scale,
+            false,
+            None,
+            None,
+        )?;
+        let attention = attention.flatten_from(candle::D::Minus2)?;
+
+        self.o_proj.forward(&attention)
+    }
+}
+
+struct Qwen3MLP {
+    gate_up_proj: Linear,
+    down_proj: Linear,
+
+    act: HiddenAct,
+    intermediate_size: usize,
+
+    span: tracing::Span,
+}
+
+impl Qwen3MLP {
+    pub fn load(vb: VarBuilder, config: &Qwen3Config) -> Result<Self> {
+        let intermediate_size = config.intermediate_size;
+
+        let gate_proj_weight = vb
+            .pp("gate_proj")
+            .get((intermediate_size, config.hidden_size), "weight")?;
+
+        let up_proj_weight = vb
+            .pp("up_proj")
+            .get((intermediate_size, config.hidden_size), "weight")?;
+
+        let gate_up_proj_weight = Tensor::cat(&[&gate_proj_weight, &up_proj_weight], 0)?;
+        let gate_up_proj = Linear::new(gate_up_proj_weight, None, None);
+
+        let down_proj_weight = vb
+            .pp("down_proj")
+            .get((config.hidden_size, intermediate_size), "weight")?;
+        let down_proj = Linear::new(down_proj_weight, None, None);
+
+        Ok(Self {
+            gate_up_proj,
+            down_proj,
+            intermediate_size,
+            act: config.hidden_act.clone(),
+            span: tracing::span!(tracing::Level::TRACE, "mlp"),
+        })
+    }
+
+    pub fn forward(&self, hidden_states: &Tensor) -> Result<Tensor> {
+        let _enter = self.span.enter();
+
+        let gate_up_states = self.gate_up_proj.forward(hidden_states)?;
+        let gate_states = gate_up_states.narrow(1, 0, self.intermediate_size)?;
+        let up_states = gate_up_states.narrow(1, self.intermediate_size, self.intermediate_size)?;
+
+        let gate_states = match self.act {
+            HiddenAct::Gelu => gate_states.gelu(),
+            HiddenAct::Relu => gate_states.relu(),
+            HiddenAct::Swiglu => gate_states.silu(),
+        }?;
```

Comment:
nit) I've recently added `forward` method to `HiddenAct` enum [here](https://github.com/huggingface/text-embeddings-inference/blob/main/backends/candle/src/layers/linear.rs#L15)!

how about using this?